### PR TITLE
refactor/gui_launcher

### PR DESCRIPTION
**Pull request by JarbasAl**
this PR decouples the GUI protocol from the enclosure service, the GUI is fully independent from the hardware integration and does not belong there imho

a new gui service is added that fully integrates the GUI protocol logic in a standalone fashion. 

Moving forward this allows us to work in PHAL independently from the mycroft HAL spec while ensuring compatibility, any mycroft enclosure importing the base enclosure class will no longer conflict when used under ovos-core

This effectively makes the enclosure service optional, it should only be used in the mark_1 after this PR

### CODE REVIEW

**Review by NeonDaniel — APPROVED** [review]
I've been thinking about this in terms of NeonCore, should gui be pulled out into a separate module? It's the most optional module in core since some installations won't implement it at all.

**Comment by JarbasAl** [comment]
> I've been thinking about this in terms of NeonCore, should gui be pulled out into a separate module? It's the most optional module in core since some installations won't implement it at all.

in the context of neon-core microservice architecture i think it would make sense to basically have the `__main__.py` file added in this PR in a new module with ovos-core as a dependency

I assume the neon enclosure service will also be replaced with PHAL when ovos implements it

File: start-mycroft.sh [hunk 1]
```diff
@@ -105,7 +107,7 @@ function launch-all() {
     launch-background skills
     launch-background audio
     launch-background voice
-    launch-background enclosure
+    launch-background gui
```
Comment (krisgesling) [inline start-mycroft.sh:110]: Is this intentionally replacing enclosure with gui rather than adding another line?

**Review by krisgesling — COMMENTED** [review]
Love this change - just curious what's the PHAL?

File: start-mycroft.sh (hunk 1 above)
Comment (JarbasAl) [reply start-mycroft.sh:110]: yes its intentional, the aim is that enclosure will be fully deprecated, the generic enclosure only purpose was to launch the gui protocol, with thi change launching enclosure is doing nothing except take up resources, it just sits there doing nothing...

the only circumstance you want to launch enclosure is in a mk1, and we generally do not encourage using these scripts but rather to create systemd services or equivalent

**Comment by JarbasAl** [comment]
> Love this change - just curious what's the PHAL?

Plugin based Hardware Abstraction Layer

**Pull request merged**